### PR TITLE
runtime: make improvements to `global_queue_interval`

### LINKED ISSUE #5713: rt: consider having `runtime::Builder` panic if setting `global_queue_interval` to zero
Doing this has the effect of inverting the order in which the multi-threaded runtime checks the global queue. This is probably not the intended behavior.

### PR DESCRIPTION
**Pull request by mox692**
## Motivation
Resolve #5713. Currently specifying 0 or 1 for `global_queue_interval` causes a panic.

## Solution
This PR:
* Added documentation indicating that setting global_queue_interval to 0 is not allowed and included an assert in the builder.
* Removed the `debug_assert!(next > 1)` that could be triggered when global_queue_interval is set to 1.
  * Although it's still unclear whether there is a practical benefit to always prioritize checking the global queue first, but i guess it's probably better than causing a panic in debug builds.

### CODE REVIEW

File: tokio/src/runtime/builder.rs [hunk 1]
```diff
@@ -769,6 +773,7 @@ impl Builder {
     /// # }
     /// ```
     pub fn global_queue_interval(&mut self, val: u32) -> &mut Self {
+        assert!(val > 0, "global_queue_interval must be greater than 0");
```
Comment (Darksonn) [inline tokio/src/runtime/builder.rs:776]: Can you add `#[track_caller]` and a [test for the panic location](https://github.com/tokio-rs/tokio/blob/master/tokio/tests/rt_panic.rs)?

File: tokio/src/runtime/scheduler/multi_thread/worker.rs [hunk 2]
```diff
@@ -985,8 +985,6 @@ impl Core {
             .stats
             .tuned_global_queue_interval(&worker.handle.shared.config);
 
-        debug_assert!(next > 1);
```
Comment (Darksonn) [inline tokio/src/runtime/scheduler/multi_thread/worker.rs:988]: It sounds like this was triggered by setting it to 1? In that case, let's add a regression test for that case.

**Review by Darksonn — APPROVED** [review]

File: tokio/tests/rt_threaded_alt.rs [hunk 3]
```diff
@@ -487,6 +487,31 @@ fn max_blocking_threads_set_to_zero() {
...
+    let cnt = Arc::new(AtomicUsize::new(0));
+    rt.block_on(async {
+        for _ in 0..10 {
+            let cnt = cnt.clone();
+            tokio::spawn(async move { cnt.fetch_add(1, Relaxed) })
+                .await
+                .unwrap();
+        }
+    });
+    assert_eq!(cnt.load(Relaxed), 10);
```
Comment (Darksonn) [inline tokio/tests/rt_threaded_alt.rs:512]: You are using `.await` on the tasks immediately when spawning, so you run them one-at-the-time. Is that what you intended?

Comment (mox692) [reply tokio/tests/rt_threaded_alt.rs:512]: Indeed, running tasks in parallel would be better here.

**Review by Darksonn — APPROVED** [review]

**Pull request merged**